Improve clarity of precedence when command-line parameters are used.

jamezpolley opened the pull request:
* Add command-line values into the precedence list.
* Several sample config snippets were included without any explanation
  of how those snippets would be processed. Added descriptions so that
  the reader can understand what each snippet will (or won't) accomplish.

##### SUMMARY
Clarify some of the docs around variable precedence by explaining the code snippets being used.

##### ISSUE TYPE
 - Docs Pull Request

jamezpolley committed:
Improve clarity of precedence when command-line parameters are used.

* Add command-line values into the precedence list.
* Several sample config snippets were included without any explanation
  of how those snippets would be processed. Added descriptions so that
  the reader can understand what each snippet will (or won't) accomplish.

gundalow commented:
There was an issue that got introduced into devel during the test update. I've triggered a rerun of CI which should pass.

bcoca commented on docs/docsite/rst/user_guide/playbooks_variables.rst:
```diff
@@ -886,22 +887,28 @@ Basically, anything that goes into "role defaults" (the defaults folder inside t
 
-This will still connect as ``ramon`` because ``ansible_ssh_user`` is set to ``ramon`` in inventory for myhost.
-For plays/tasks this is also true for ``remote_user``::
+This will still connect as ``ramon`` because the value from the inventory takes priority.
```
this is slightly incorrect, the value of the variable takes precedence, it does not matter if it was defined in inventory or elsewhere. This can mislead users to think you can ONLY define this var in inventory, it must be made clear that is is the fact that a connection variable exists, no matter where it is defined in.

jamezpolley replied:
*finally notices response* 

Sorry for the delay. Let me try something different..

jamezpolley replied:
I've updated the wording here to try to make this more clear.

jamezpolley committed:
Don't focus on inventory as much

Expand on the fact that it's the fact that a variable is set that
matters, not the source of the variable.

samccann commented:
Thanks for helping improve the clarity of Ansible documentation!
